Challenge-4/Bot_1/src/agent.spec.ts

```diff
@@ -0,0 +1,284 @@
+  FindingType,
+  HandleTransaction,
+} from "forta-agent";
+import { ASSET_INFO, CONFIGURATOR_PROXY } from "./constants";
+import * as networkManager from "./networkManager";
+
+describe("Compound test suite", () => {
+  let handleTransaction: HandleTransaction;
+  let mockProvider: MockEthersProvider;
+  let txEvent: TestTransactionEvent;
+  const mockAssetTokenAddress = createAddress("0x10957");
+  const mockArgs = [
+    createAddress("0x1421"),
+    createAddress("0x123"),
+    createAddress("0x234"),
+  ];
+  const Iface = new ethers.utils.Interface(ASSET_INFO);
+  const getCollateralName = new ethers.utils.Interface([
+    "function name() view returns (string)",
+  ]);
+
+  const setupMockProvider = (collateralAddresses: string[]) => {
+    mockProvider.setNetwork(1);
+    collateralAddresses.forEach((address, index) => {
+      mockProvider.addCallTo(mockAssetTokenAddress, 0, Iface, "getAssetInfo", {
+        inputs: [index],
+        outputs: [
+          index,
+          address,
+          createAddress("0x234"),
+          4,
+          12,
+          134,
+          1245,
+          1265,
+        ],
+      });
+      mockProvider.addCallTo(address, 0, getCollateralName, "name", {
+        inputs: [],
+        outputs: [`Collateral-${index + 1}`],
+      });
+    });
+  };
+
+  beforeEach(() => {
+    mockProvider = new MockEthersProvider() as any;
+    const provider = mockProvider as unknown as ethers.providers.Provider;
+    handleTransaction = provideHandleGovernanceTransaction(
+      provider,
+      ASSET_INFO,
+    );
+    txEvent = new TestTransactionEvent().setBlock(0);
+    jest
+      .spyOn(networkManager, "getAddress")
+      .mockResolvedValue(mockAssetTokenAddress);
+    jest
+      .spyOn(networkManager, "getConfigurator")
+      .mockResolvedValue(CONFIGURATOR_PROXY);
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it("should return single finding for single event occurence", async () => {
+    const collateralAddresses = [
+      createAddress("0x1247"),
+      createAddress("0x12567"),
+      createAddress("0x1289"),
+      createAddress("0x1290"),
+    ];
+    setupMockProvider(collateralAddresses);
+
+    txEvent.addEventLog(
+      "event SetGovernor(address indexed cometProxy, address indexed oldGovernor, address indexed newGovernor)",
+      CONFIGURATOR_PROXY,
+      [...mockArgs],
+    );
+
+    const metadata1: { [key: string]: any } = {
+      SET_GOVERNOR: {
+        Old_value: createAddress("0x123"),
+        New_value: createAddress("0x234"),
+      },
+    };
+    const metadata = {
+      "Collateral Asset - Collateral-1":
+        "0x0000000000000000000000000000000000001247",
+      "Collateral Asset - Collateral-2":
+        "0x0000000000000000000000000000000000012567",
+      "Collateral Asset - Collateral-3":
+        "0x0000000000000000000000000000000000001289",
+      "Collateral Asset - Collateral-4":
+        "0x0000000000000000000000000000000000001290",
```

Comment:
```suggestion
      "Collateral Asset - Collateral-1":
        `${createAddress("0x1247")},
      "Collateral Asset - Collateral-2":
        "0x0000000000000000000000000000000000012567",
      "Collateral Asset - Collateral-3":
        "0x0000000000000000000000000000000000001289",
      "Collateral Asset - Collateral-4":
        "0x0000000000000000000000000000000000001290",
```
These could be updated so that they use formatted strings. Only updated the first one as an example.

Reply:
But also, if you set a variable to `createAddress("0x1247")`, then you could use _that_ in both L77 and L98, so that if you need to change it, you only do so in one location.